### PR TITLE
(GH185) Use more descriptive default build file name of psakefile.ps1

### PR DESCRIPTION
**Pull request by glennsarti**
Resolves https://github.com/psake/psake/issues/185

The default psake build file of default.ps1 is completely ambiguous without
inspecting its contents. Rake has a Rakefile, bundle has a Gemfile etc. so
instead, psake should use Psakefile.ps1 as the default build file name.

This PR modifies the load behaviour for a default build file:

- Refactored detecting the default build filename into it's own function

- The `Get-DefaultBuildFile` will prefer to load the `buildFileName` file.  If
it does not exist it fall back to `legacyBuildFileName` and emit a deprecation
warning if it exists.  This allows all existing users to continue to use
`default.ps1` and migrate to psakefile.ps1 when appropriate.

- Adds tests for the behaviour

- Updates documentation with the new behaviour

### CODE REVIEW

**Comment by glennsarti** [comment]
*bump*

**Comment by gep13** [comment]
@glennsarti I haven't forgotten about this, just working my way through the todo list 😢

**Comment by glennsarti** [comment]
sok...I'll just keep bumping it :-)

**Comment by devblackops** [comment]
I like it @glennsarti!

File: README.md [hunk 1]
```diff
@@ -1,8 +1,9 @@
-﻿# Welcome to the psake project
+Welcome to the psake project
+=============================
 
-[![Build status][appveyor-badge]][appveyor-build]
-[![Build Status][travis-badge]][travis-build]
-[![Join the chat at https://gitter.im/psake/psake][gitter-badge]][gitter]
+[![Build status](https://ci.appveyor.com/api/projects/status/e8b90u1q1ex5hx9m?svg=true)](https://ci.appveyor.com/project/psake/psake)
```
Comment (rkeithhill) [inline README.md]: We lost the changes I made on this one.  I think we could just copy/paste the top and bottom of the file I committed to this one and we'd have it.

Comment (rkeithhill) [reply README.md]: Or if it would be easier, you can go ahead and get this committed and I can resubmit another PR to put back my changes.

Comment (devblackops) [reply README.md]: My fault. I'll add it back.

**Comment by glennsarti** [comment]
Is this the best place to be doing this in? i.e. Shouldn't these be in separate PRs? or at least on your own forks?

**Comment by devblackops** [comment]
Yep. Just fixing merge conflicts since there was a flurry of changes last night.

**Pull request merged**